pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator_test.go

```diff
@@ -0,0 +1,197 @@
+						},
+					},
+				},
+			},
+			interfaces: []*compute.NetworkInterface{
+				{
+					AliasIpRanges: []*compute.AliasIpRange{
+						{IpCidrRange: "10.11.1.0/24", SubnetworkRangeName: DefaultSecondaryRangeA},
+					},
+					NetworkIP:  "80.1.172.1",
+					Network:    DefaultVPCName,
+					Subnetwork: DefaultVPCSubnetName,
+				},
+			},
+			wantDefaultNwPodCIDRs: []string{"10.11.1.0/24"},
+		},
+		{
+			desc: "one additional network along with default network",
+			networks: []*networkv1.Network{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: networkv1.DefaultNetworkName,
+					},
+					Spec: networkv1.NetworkSpec{
+						Type: "L3",
+						ParametersRef: &networkv1.NetworkParametersReference{
+							Group: Group,
+							Kind:  GKENetworkParamsKind,
+							Name:  DefaultGKENetworkParamsName,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: RedNetworkName,
+					},
+					Spec: networkv1.NetworkSpec{
+						Type: "L3",
+						ParametersRef: &networkv1.NetworkParametersReference{
+							Group: Group,
+							Kind:  GKENetworkParamsKind,
+							Name:  RedGKENetworkParamsName,
+						},
+					},
+				},
+			},
+			gkeNwParams: []*networkv1.GKENetworkParams{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: DefaultGKENetworkParamsName,
+					},
+					Spec: networkv1.GKENetworkParamsSpec{
+						VPC:       DefaultVPCName,
+						VPCSubnet: DefaultVPCSubnetName,
+						PodIPv4Ranges: &networkv1.SecondaryRanges{
+							RangeNames: []string{DefaultSecondaryRangeA, DefaultSecondaryRangeB},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: RedGKENetworkParamsName,
+					},
+					Spec: networkv1.GKENetworkParamsSpec{
+						VPC:       RedVPCName,
+						VPCSubnet: RedVPCSubnetName,
+						PodIPv4Ranges: &networkv1.SecondaryRanges{
+							RangeNames: []string{RedSecondaryRangeA, RedSecondaryRangeB},
+						},
+					},
+				},
+			},
+			interfaces: []*compute.NetworkInterface{
+				{
+					AliasIpRanges: []*compute.AliasIpRange{
+						{IpCidrRange: "10.11.1.0/24", SubnetworkRangeName: DefaultSecondaryRangeA},
+					},
+					NetworkIP:  "80.1.172.1",
+					Network:    DefaultVPCName,
+					Subnetwork: DefaultVPCSubnetName,
+				},
+				{
+					AliasIpRanges: []*compute.AliasIpRange{
+						{IpCidrRange: "172.11.1.0/24", SubnetworkRangeName: RedSecondaryRangeA},
+					},
+					NetworkIP:  "10.1.1.1",
+					Network:    RedVPCName,
+					Subnetwork: RedVPCSubnetName,
+				},
+			},
+			wantDefaultNwPodCIDRs: []string{"10.11.1.0/24"},
+			wantNorthInterfaces: networkv1.NorthInterfacesAnnotation{
+				{
+					Network:   RedNetworkName,
+					IpAddress: "10.1.1.1",
+				},
+			},
+			wantAdditionalNodeNetworks: networkv1.MultiNetworkAnnotation{
+				{
+					Name:  RedNetworkName,
+					Scope: "host-local",
+					Cidrs: []string{"172.11.1.0/24"},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		objects := make([]runtime.Object, 0)
```

Comment:
better to use subtests with `t.Run` to avoid test pollution

Reply:
Done.